packages/file-explorer/src/FileExplorer.tsx

```diff
@@ -0,0 +1,162 @@
+
+    useEffect(() => {
+      let tablePromise: CancelablePromise<FileStorageTable>;
+      async function initTable() {
+        log.debug('initTable');
+
+        tablePromise = PromiseUtils.makeCancelable(storage.getTable(), t =>
+          t.close()
+        );
+
+        try {
+          setTable(await tablePromise);
+        } catch (e) {
+          if (!PromiseUtils.isCanceled(e)) {
+            log.error('Unable to initialize table', e);
+          }
+        }
+      }
+      initTable();
+      return () => {
+        tablePromise.cancel();
+      };
+    }, [storage]);
+
+    const handleError = useCallback((e: Error) => {
+      if (!PromiseUtils.isCanceled(e)) {
+        log.error(e);
+      }
+    }, []);
+
+    const handleDelete = useCallback((files: FileListItem[]) => {
+      log.debug('handleDelete, pending confirmation', files);
+      setItemsToDelete(files);
+    }, []);
+
+    const handleDeleteConfirm = useCallback(() => {
+      log.debug('handleDeleteConfirm', itemsToDelete);
+      itemsToDelete.forEach(file =>
+        storage.deleteFile(
+          isDirectory(file) ? FileUtils.makePath(file.filename) : file.filename
+        )
+      );
+      onDelete(itemsToDelete);
+      setItemsToDelete([]);
+    }, [itemsToDelete, onDelete, storage]);
+
+    const handleDeleteCancel = useCallback(() => {
+      log.debug('handleDeleteCancel');
+      setItemsToDelete([]);
+    }, []);
+
+    const handleRename = useCallback(
+      (item: FileListItem, newName: string) => {
+        let name = item.filename;
+        const isDir = isDirectory(item);
+        if (isDir && !name.endsWith('/')) {
+          name = `${name}/`;
+        }
+        let destination = `${FileUtils.getParent(name)}${newName}`;
+        if (isDir && !destination.endsWith('/')) {
+          destination = `${destination}/`;
+        }
+        log.debug2('handleRename', name, destination);
+        storage.moveFile(name, destination).catch(handleError);
+        onRename(name, destination);
+      },
+      [handleError, onRename, storage]
+    );
+
+    useImperativeHandle(ref, () => ({
+      updateDimensions: () => {
+        fileListContainer.current?.updateDimensions();
+      },
+    }));
+
+    const isDeleteConfirmationShown = itemsToDelete.length > 0;
+    const deleteConfirmationMessage = useMemo(() => {
+      if (itemsToDelete.length === 1) {
+        return `Are you sure you want to delete "${itemsToDelete[0].itemName}"?`;
+      }
+      return `Are you sure you want to delete the selected files?`;
+    }, [itemsToDelete]);
+
+    return (
+      <div className="file-explorer">
+        {table && (
+          <FileListContainer
+            ref={fileListContainer}
+            isMultiSelect={isMultiSelect}
+            showContextMenu
+            onDelete={handleDelete}
+            onRename={handleRename}
+            onSelect={onSelect}
+            rowHeight={rowHeight}
+            table={table}
```

Comment:
Needs `validateRename` prop for the inline editor, just add a TODO here for now.